baybe/recommenders/base.py

```diff
@@ -106,16 +105,29 @@ def recommend(
             batch_size: The number of points that should be recommended.
             train_x: The training data used to train the model.
             train_y: The training labels used to train the model.
-            allow_repeated_recommendations: Allow to make recommendations that were
-                already recommended earlier. This only has an influence in discrete
-                search spaces.
-            allow_recommending_already_measured: Allow to output recommendations that
-                were measured previously. This only has an influence in discrete
-                search spaces.
 
         Returns:
             A DataFrame containing the recommendations as individual rows.
         """
+        ...
+
+
+@define
+class Recommender(ABC, RecommenderProtocol):
```

Comment:
According to the intended usage of `Protocol`, it should not be necessary to use inheritance here. This should be double-checked!

Reply:
Potentially related: The init function now has a really weird signature, both in the table and in the overview:
![image](https://github.com/emdgroup/baybe/assets/84016733/ce4bb096-7e06-419d-8d8d-2119d92a744a)
![image](https://github.com/emdgroup/baybe/assets/84016733/a4a193ba-57bc-4ade-a62a-dc345b4ab12c)
We should just check that this is fine for us resp. that we might want to remove the function from e.g. the table.

Reply:
keeping inheritance as discussed in dev chat.

regarding the doc: i think this is partially because they are kw_only. I tried the look also without kw_only but that just removes the `*` and it mostly looks the same

to me, the init doesnt look that weird overall? Also, these should really be kw_only and changing back due to this minor thing imos doesnt make sense, hence ill resolve